addons/webhooks/clusterbootstrap_webhook.go

```diff
@@ -292,7 +292,12 @@ func (wh *ClusterBootstrap) getGVR(gk schema.GroupKind) (*schema.GroupVersionRes
 	}
 	apiResourceList, err := wh.cachedDiscoveryClient.ServerPreferredResources()
 	if err != nil {
-		return nil, err
+		if apiResourceList == nil {
```

Comment:
I think we can skip this check since we do iterate over apiResourceList further down the code. and if apiresourceList == nil, we return an error there.